rfcs/text/0000-project-ACL.md

```diff
@@ -0,0 +1,120 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# Resource Based Access Control
+
+## Summary
+
+The DCP needs the ability to control access to projects across the different components in the DCP. This requires a 
+centralized source of truth on who has access and an agreed process for managing access. This RFC seeks to unify the 
+DCP’s storage and evaluation of project ACL using the described implementation.
+
+## Author(s)
+
+`[Trent Smith](mailto:trent.smith@chanzuckerberg.com)`
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+- Required to support [HCA Project Complete Stage RFC](https://github.com/HumanCellAtlas/dcp-community/blob/3ce2d4e786432d16a8540da599fb2f72a89f8c5c/rfcs/text/0015-project-completion-stage.md)
+- Required to support Project Bases Access Control
+- To limit the users view of the DCP to only the projects they have access.
+- Allow storage of controlled access data that meets FISMA and GDPR access control requirement.
+
+### User Stories
+
+1. See user stories in [HCA Project Complete Stage RFC](https://github
+.com/HumanCellAtlas/dcp-community/blob/3ce2d4e786432d16a8540da599fb2f72a89f8c5c/rfcs/text/0015-project-completion-stage.md).
+
+1. As a data submitter, I need to limit access to my project, so that I can't ensure it is version-complete before 
+ making public.
+
+1. As a data submitter, I would like to give my lab access to my project, so they can assist in validating secondary 
+ analysis results.
+
+1. As a data submitter, I would like to make my private project accessible to the public, so that other user of the DCP
+ can benefit.  
+
+1. As a data submitter, I would like to submit data that has personally identifiable information (PII) and therefore 
+ need special permissions to access.
+ 
+1. As a component of the DCP, I would like to limit a user's view to only resource they have permission to view, to 
+ prevent exposing privileged information to a users.
+
+1. As an operator of the DCP, I would like to know why a user has access to a resource, so I may reason about that 
+ whether a user is over privileged. This is helpful for maintaining a access policy of least privileges.
+
+*Share the [User Stories](https://www.mountaingoatsoftware.com/agile/user-stories) motivating this RFC.*
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
```

Comment:
I don't see any mention of the Data Browser index (Azul) and if/how it would be expected to restrict access to metadata for projects. I don't think this RFC is necessary to implement support for the Project Completion Stage RFC (where access could be restricted at the level of "Released View" vs "Unreleased View"), @hannes-ucsc and @NoopDog should comment further.

Reply:
I agree, seeing as this is collaborative effort I invite anyone from Azul to help determine these requirements. If i were take a stab at it, I would say the databrowser would check with either fusillade or query service to list all project the user has access to and limit the users view as such.

I believe part of the requirements of the Project Completion Stage RFC was to limit the users view to only project that have been released, as well as the ability for a user to release their project once final validation has occurred.